Tmp values on script without main

**V version:** 
V 0.1.10

**What did you do?**
```
mut m := map[string]int{}
m['one'] = 1
m['two'] = 2
key := 'test'
println(m[key])
```

**What did you see ?**
```
/home/Henrixounez//.vlang//test2.c:3161:36: error: ‘m’ undeclared here (not in a function); did you mean ‘tm’?
 int tmp5 = 0; bool tmp6 = map_get( m , key, & tmp5);
                                    ^
                                    tm
/home/Henrixounez//.vlang//test2.c:3161:40: error: ‘key’ undeclared here (not in a function); did you mean ‘key_t’?
 int tmp5 = 0; bool tmp6 = map_get( m , key, & tmp5);
                                        ^~~
                                        key_t
V panic: clang error
```



**What is happening ?**

The generated main :
```
int main() {
  map_int m= new_map(1, sizeof(int)) ;
  map__set(& m , tos2("one") , & tmp2) ;
  map__set(& m , tos2("two") , & tmp3) ;
  string key= tos2("test") ;
  println (int_str( tmp5 ) ) ;
  ;
  return 0;
}
```

The problem is that tmp values are created outside of generated main:
```
int tmp2 =  1;
int tmp3 =  2;
int tmp5 = 0; bool tmp6 = map_get( m , key, & tmp5);
```

--- a/compiler/parser.v
+++ b/compiler/parser.v
@@ -259,8 +259,11 @@ fn (p mut Parser) parse() {
 						p.cur_fn.clear_vars()
 					}
 				} 
-				start := p.cgen.lines.len
+				mut start := p.cgen.lines.len
 				p.statement(true)
+				if p.cgen.lines[start - 1] != '' && p.cgen.fn_main != '' {
+					start--
+				}
 				p.genln('') 
 				end := p.cgen.lines.len
 				lines := p.cgen.lines.slice(start, end)

